### DISCUSSION TITLE
VMXTranscriber error for Voicemail

### DISCUSSION THREAD

**Discussion by shaaju** [amazon-connect/amazon-connect-salesforce-scv#159] [Q&A]
Last year, I implemented **VMX2-VoicemailExpress**. It was working fine until we updated the Contact Center on Salesforce and updated certificates as well for Authentication. After the update, we noticed that some Voicemails are going through as Case on salesforce while others are not even though an audio file is created in S3 bucket. Upon further investigation, I have found that the **VMXTranscriber function** is giving following error in logs:

[DEBUG]	2023-08-10T16:43:38.883Z	773a8487-2f8c-4ff6-ae55-5f45d2f40bc7	Response body:b'{     "__type": "ConflictException",     "Message": "**The requested job name already exists. Use a different job name**." }'

[DEBUG]	2023-08-10T16:43:38.884Z	773a8487-2f8c-4ff6-ae55-5f45d2f40bc7	Response headers: {'x-amzn-RequestId': '3a06d5b4-3f08-4136-b0fa-6b476daa16e7', 'Content-Type': 'application/x-amz-json-1.1', 'Content-Length': '107', 'Date': 'Thu, 10 Aug 2023 16:43:38 GMT', 'Connection': 'close'}

[DEBUG]	2023-08-10T16:43:38.884Z	773a8487-2f8c-4ff6-ae55-5f45d2f40bc7	Response body:b'{     "__type": "ConflictException",     "Message": "The requested job name already exists. Use a different job name." }'

[DEBUG]	2023-08-10T16:43:38.884Z	773a8487-2f8c-4ff6-ae55-5f45d2f40bc7	Event needs-retry.transcribe.StartTranscriptionJob: calling handler <botocore.retryhandler.RetryHandler object at 0x7f725f582f10>

[DEBUG]	2023-08-10T16:43:38.884Z	773a8487-2f8c-4ff6-ae55-5f45d2f40bc7	No retry needed.

**[ERROR]	2023-08-10T16:43:38.884Z	773a8487-2f8c-4ff6-ae55-5f45d2f40bc7	An error occurred (ConflictException) when calling the StartTranscriptionJob operation: The requested job name already exists. Use a different job name.**

[DEBUG]	2023-08-10T16:43:38.884Z	773a8487-2f8c-4ff6-ae55-5f45d2f40bc7	Record 1 Result: Transcription job failed

**Comment by dougjaso** [comment]
When you updated your contact center, did you also select the Voicemail option from Salesforce? Since Salesforce mostly replicated the same process for their implementation of voicemail, it is possible that their functions are running at the same time and causing an issue. If not, the easiest way to fix this specific error is to delete any stuck transcribe jobs and retest.

Transcriber takes place before any attempt to login to Salesforce takes place, so changes to certificates would not impact that function.

**Comment by dougjaso** [comment]
Discovered a new issue here. If you are using Voicemail Express, and have updated your SCV deployment to Spring 23 or newer, the integrated voicemail option for SCV (even if not configured) will conflict with VMX. This effective means that you have two complete voicemail solutions trying to handle the same voicemail, which can result in this error. While you should absolutely migrate to the native voicemail option for SCV as soon as you are ready, in the interim, you can fix this by removing the trigger from the Salesforce provided %instance%-VoiceMailAudioProcessingFunction. This effectively disables the built-in voicemail function and removes the conflict. Please note the trigger configuration. Once you are ready to migrate from VMX to native voicemail, reconfigure the trigger and remove VMX. Uninstall instructions will be provided for VMX by Oct 1 2023.
